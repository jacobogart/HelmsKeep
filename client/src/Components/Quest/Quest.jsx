import React, { Component } from 'react';

export class Quest extends Component {
	constructor(props) {
		super(props);
		this.state = {
			showCompleted: false
		};
	}

	toggleShowCompleted = () => {
		this.setState({ showCompleted: !this.state.showCompleted });
	};

	render() {
		const { title, tasks } = this.props.data;
		const completedTaskItems = [];
		const uncompletedTaskItems = [];

		tasks.forEach(task => {
      const taskItem = <li className="challenge-txt" key={task.id} contentEditable="true">{task.message}</li>;
			task.isCompleted ? completedTaskItems.push(taskItem) : uncompletedTaskItems.push(taskItem);
		});

		return (
      <article className="Quest">
        <div className="card-header">
          <h2 className="card-title" contentEditable="true">{title}</h2>
        </div>
        <div className="card-body">
          <ul>{uncompletedTaskItems}</ul>
          <p className="showCompleted" onClick={this.toggleShowCompleted}>
            Show {completedTaskItems.length} completed challenges
          </p>
          <ul>{this.state.showCompleted && completedTaskItems}</ul>
          <button className="delete-btn" type="button">
            <i className="fas fa-trash"></i>
          </button>
        </div>
			</article>
		);
	}
}

export default Quest;
