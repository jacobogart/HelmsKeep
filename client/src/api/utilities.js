export const host = 'http://localhost:3000/';

export const mockTask = {
  id: 1234,
  isCompleted: false, 
  message: 'Clean the kitchen'
}

export const mockNote = {
  id: 1234,
  title: 'My First Quest',
  tasks: [mockTask, mockTask]
}