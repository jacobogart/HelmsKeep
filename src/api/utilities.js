export const host = 'http://localhost:3000/';

export const mockTask = {
  id: 1234,
  isCompleted: true, 
  message: 'Clean the kitchen'
}

export const mockNote = {
  id: 1234,
  tasks: [mockTask, mockTask]
}