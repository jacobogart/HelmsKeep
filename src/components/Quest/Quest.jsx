import React, { Component } from 'react';

export class Quest extends Component {
	constructor(props) {
		super(props);
		this.state = {
			showCompleted: false
		};
	}

	toggleShowCompleted = () => {
		this.setState({ showCompleted: !this.state.showCompleted });
	};

	render() {
    const { title, challenges } = this.props.data;
		const completedTaskItems = [];
		const uncompletedTaskItems = [];

    // const challengeList = challenges.map((chal, i) => {
    //   return <li className="challenge-txt" key={i} contentEditable="true">{chal}</li>
    // })

		return (
      <article className="Quest">
        <div className="card-header">
          <h2 className="card-title" contentEditable="true">{title}</h2>
        </div>
        <div className="card-body">
          <ul>{challenges}</ul>
          {/* <ul>{uncompletedTaskItems}</ul>
          <p className="showCompleted" onClick={this.toggleShowCompleted}>
            Show {completedTaskItems.length} completed challenges
          </p>
          <ul>{this.state.showCompleted && completedTaskItems}</ul> */}
          <button className="delete-btn" type="button">
            <i className="fas fa-trash"></i>
          </button>
        </div>
			</article>
		);
	}
}

export default Quest;
