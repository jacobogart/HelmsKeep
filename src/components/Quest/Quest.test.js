import React from './node_modules/react';
import { shallow } from './node_modules/ enzyme';
import Quest from './Quest';
import { mockNote } from '../../api/utilities';

describe('Quest', () => {
	let wrapper, instance;

	beforeEach(() => {
		wrapper = shallow(<Quest data={mockNote} />);
		instance = wrapper.instance();
	});

	it('should match snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should have a default state', () => {
		const defaultState = { showCompleted: false };
		expect(wrapper.state()).toEqual(defaultState);
	});

	describe('Event Handlers', () => {
		it('should invoke toggleShowCompleted on click', () => {
			expect(wrapper.state('showCompleted')).toEqual(false);
			wrapper.find('.showCompleted').simulate('click');
			expect(wrapper.state('showCompleted')).toEqual(true);
		});
	});

	describe('toggleShowCompleted', () => {
		it('should toggle the showCompleted property in state', () => {
			expect(wrapper.state('showCompleted')).toEqual(false);
			instance.toggleShowCompleted();
			expect(wrapper.state('showCompleted')).toEqual(true);
			instance.toggleShowCompleted();
			expect(wrapper.state('showCompleted')).toEqual(false);
		});
	});
});
