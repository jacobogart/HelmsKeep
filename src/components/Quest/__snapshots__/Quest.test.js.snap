// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quest should match snapshot 1`] = `
<article>
  <h2>
    My First Quest
  </h2>
  <ul>
    <li
      key="1234"
    >
      Clean the kitchen
    </li>
    <li
      key="1234"
    >
      Clean the kitchen
    </li>
  </ul>
  <p
    className="showCompleted"
    onClick={[Function]}
  >
    Show 
    0
     completed challenges
  </p>
  <ul />
</article>
`;
